common/interfaces/flash_interface/flash_api.c

```diff
@@ -728,8 +729,18 @@ int update_wallet_locked_flash(const char *name, const bool is_wallet_locked) {
 
   flash_wallet->is_wallet_locked = is_wallet_locked;
 
-  // Reset nonce to all zero
-  memset(flash_wallet->challenge.nonce, 0, POW_NONCE_SIZE);
+  if (is_wallet_locked) {
```

Comment:
add a comment reasoning behind this change.

Reply:
updated